tools/driver/driver.cpp

```diff
@@ -159,6 +160,33 @@ static int run_driver(StringRef ExecName,
   DiagnosticEngine Diags(SM);
   Diags.addConsumer(PDC);
 
+  if (auto useSwiftDriver =
+      llvm::sys::Process::GetEnv("SWIFT_USE_NEW_DRIVER")) {
+    SmallString<256> NewDriverPath(llvm::sys::path::parent_path(Path));
+    llvm::sys::path::append(NewDriverPath, "swift-driver");
+    SmallVector<const char *, 256> subCommandArgs;
+    // Rewrite the program argument.
+    subCommandArgs.push_back(NewDriverPath.c_str());
+    if (ExecName == "swiftc") {
+      subCommandArgs.push_back("--driver-mode=swiftc");
+    } else {
+      assert(ExecName == "swift");
```

Comment:
There are other symlinks to the C++ driver, such as `swift-symbolgraph-extract`. Rather than assert that it is `swift` or `swiftc`, can we please check only respect `SWIFT_USE_NEW_DRIVER` when `ExecName` is one of those two? Rather than asserting?

Reply:
Sounds reasonable, will fix.